bftengine/tests/bcstatetransfer/RVT_test.cpp

```diff
@@ -129,313 +207,313 @@ TEST_P(RVTHashTestParamFixture, basicSumAndModOps) {
+  ostringstream oss;
+  for (auto& c : output) {
+    oss << c;
+  }
+  ASSERT_EQ(oss.str(), input);
+}
 
 TEST_F(RVTTest, StartIntheMiddleInsertionsOnly) {
-  uint32_t RVT_K = 12;
-  uint32_t fetch_range_size_ = 5;
-  RangeValidationTree rvt(logger_, RVT_K, fetch_range_size_);
+  const uint32_t RVT_K = 12;
+  const uint32_t fetch_range_size = 5;
+  auto config = RVTConfig(RVT_K, fetch_range_size, 32);
+  init(config);
   // These are block Ids, the matching RVB index is 1,10,600,30
   std::vector<RVBId> start_rvb_ids{5, 50, 3000, 150};
-  string str{"11"};
 
   for (const auto& start_rvb_id : start_rvb_ids) {
-    for (RVBId i{0}; i <= 1000; ++i) {
-      rvt.addNode(start_rvb_id + i * fetch_range_size_, str.data(), str.size());
-      rvt.validate();
-    }
-    rvt.clear();
+    RVBId i = 0;
+    RVBId j = 1000;
+    ASSERT_NFF(rvt_delegator_->addNode(start_rvb_id + i * fetch_range_size, start_rvb_id + j * fetch_range_size));
+    ASSERT_NFF(rvt_delegator_->clear());
   }
 }
 
-TEST_F(RVTTest, constructTreeWithSingleFirstNode) {
-  static constexpr uint32_t fetch_range_size = 4;
-  RangeValidationTree rvt(logger_, RVT_K, fetch_range_size);
-  for (auto i = fetch_range_size; i <= fetch_range_size; i = i + fetch_range_size) {
-    STDigest digest(std::to_string(i).c_str());
-    rvt.addNode(i, digest.get(), BLOCK_DIGEST_SIZE);
-    ASSERT_TRUE(rvt.validate());
-  }
-  ASSERT_EQ(rvt.totalNodes(), 1);
-  ASSERT_EQ(rvt.empty(), false);
-  ASSERT_EQ(rvt.totalLevels(), 1);
+TEST_F(RVTTest, constructSingleNodeHavingFirstChild) {
+  auto config = getRandomConfig();
+  init(config);
+  ASSERT_NFF(rvt_delegator_->addNode(config.fetch_range_size_, config.fetch_range_size_));
+  ASSERT_EQ(rvt_delegator_->empty(), false);
+  ASSERT_EQ(rvt_delegator_->totalLevels(), 1);
+  ASSERT_EQ(rvt_delegator_->totalNodes(), 1);
 }
 
 TEST_F(RVTTest, constructTreesWithSingleMiddleNode) {
-  size_t fetch_range_size = 4;
-  uint32_t RVT_K = 4;
-  RangeValidationTree rvt(logger_, RVT_K, fetch_range_size);
-
+  const size_t fetch_range_size = 4;
+  const uint32_t RVT_K = 4;
+  auto config = RVTConfig(RVT_K, fetch_range_size, 32);
+  init(config);
   for (size_t i = 100; i < 1000; ++i) {
     for (size_t j{i}; j < (i + RVT_K * 2 + 1); ++j) {
-      STDigest digest{};
-      auto str = std::to_string(i);
-      memcpy(digest.getForUpdate(), str.c_str(), str.size());
-      rvt.addNode(j * fetch_range_size, digest.get(), BLOCK_DIGEST_SIZE);
-      ASSERT_TRUE(rvt.validate());
+      rvt_delegator_->addNode(j * fetch_range_size, j * fetch_range_size);
     }
     for (size_t j{i}; j < (i + RVT_K * 2 + 1); ++j) {
-      STDigest digest{};
-      auto str = std::to_string(i);
-      memcpy(digest.getForUpdate(), str.c_str(), str.size());
-      rvt.removeNode(j * fetch_range_size, digest.get(), BLOCK_DIGEST_SIZE);
-      ASSERT_TRUE(rvt.validate());
+      rvt_delegator_->removeNode(j * fetch_range_size, j * fetch_range_size);
```

Comment:
my fault, can you please insert at the end a check that tree is empty? after each iteration of i?